File: 백준/Silver/10845. 큐/큐.py
import sys
N = int(sys.stdin.readline())
q = []
for _ in range(N):
    com = sys.stdin.readline().split()
    if com[0] == "push":
        q.insert(0,com[1])
    elif com[0] == "pop":
        if len(q) != 0:
            print(q.pop())
        else:
            print(-1)
    elif com[0] == "size":
        print(len(q))
    elif com[0] == "empty":
        if len(q) != 0:
            print(0)
        else:
            print(1)
    elif com[0] == "front":
        if len(q) == 0:
            print(-1)
        else:
            print(q[len(q)-1])
    elif com[0] == "back":
        if len(q) == 0:
            print(-1)
        else:
            print(q[0])

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        Queue<Integer> q = new LinkedList<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        int last = 0;
        int n = Integer.parseInt(st.nextToken());

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String str = st.nextToken();
            if (str.equals("push")) {
                last = Integer.parseInt(st.nextToken());
                q.add(last);
            }
            if (str.equals("pop")) {
                if (q.isEmpty()) {
                    sb.append(-1).append("\n");
                } else {
                    sb.append(q.poll()).append("\n");
                }
            }
            if (str.equals("size")) {
                sb.append(q.size()).append("\n");
            }
            if (str.equals("empty")) {
                if (q.isEmpty() == true) {
                    sb.append(1).append("\n");
                } else {
                    sb.append(0).append("\n");
                }
            }
            if (str.equals("front")) {
                if (q.isEmpty()) {
                    sb.append(-1).append("\n");
                } else {
                    sb.append(q.peek()).append("\n");
                }
            }
            if (str.equals("back")) {
                if (q.isEmpty()) {
                    sb.append(-1).append("\n");
                } else {
                    sb.append(last).append("\n");
                }
            }
        }
        System.out.println(sb);
    }
}